docu/Concepts/Snippets/1-Blockchain/Connector/Anforderungen.md

```diff
@@ -0,0 +1,51 @@
+voran zu kommen. Nur die sicherheitsrelevanten Teile blieben in C++.
+
+### Gesichert vor Hackern und Abschaltung
+- Geschützt vor einzelnen Hacker die sich ein paar Gradidos dazu tricksen wollen
+- Geschützt davor das einzelne Menschen ausgeschlossen werden können z.B. kritische Journalisten
+- Geschützt davor das es von einzelnen Admins einfach abgeschaltet werden kann (z.B. durch Bestechung)
+- Sichern der Private Keys so das kein Hacker sie selbst bei einem Server Hack in die Finger bekommen kann
+- Transparenz (das man z.B. bei Politikern sehen kann von wem sie alles Geld bekommen)
+
+Mit einem C++ Programm hat man die Möglichkeit dem Program beim Start einen Schlüssel mitzugeben der zusätzlich für die Verschlüsselung der Passwörter verwendet wird. 
+Nicht als Parameter sondern per Eingabe so das das Passwort nicht im bash log auftaucht. 
+Oder schon direkt beim compilieren und auf dem Production Server landet dann nur das compilierte Binary. Das ist aber weniger sicher, 
+weil ein Hacker das Binary untersuchen kann um den Schlüssel daraus zu extrahieren, auch wenn es mit Obfuscator Techniken erschwert werden kann.
+
+Auf die Art wäre es für einen Hacker selbst dann unmöglich an die Private Keys der Benutzer zu kommen, wenn er den Server gehackt hat,
+auf dem der Login-Server läuft und er E-Mails und Passwörter von Benutzern erbeutet hat. Dieser Schlüssel würde also nirgendwo auf dem 
+Server gespeichert werden und läge damit nur im Arbeitsspeicher vor. Dazu müsste man dann noch Techniken benutzen um zu verhindern 
+das der Schlüssel aus dem Arbeitsspeicher gelesen werden kann. 
+
+### In der Benutzung so einfach wie paypal
+Also das Gradidos einfach an eine E-Mail Addresse gesendet werden können.
+Der größte Teil der aktuellen Zielgruppe ist nicht besonders Computeraffin.
+
+### Ressourcenverbrauch
+Grundsätzlich ist ein geringer Ressourcenverbrauch von Vorteil.
+Bei der Softwareentwicklung gibt es einmal die Ressourcen Manpower die Stunden die ein Entwickler reinstecken muss
+und die Ressource Hardware wieviel CPU Leistung, Arbeitsspeicher usw. und damit auch Stromverbrauch von dem laufenden Programm benötigt wird.
+Wenn das Program nur von wenigen ab und zu mal genutzt wird, macht es nicht viel Sinn auf die Effizienz zu schauen und es reicht
+z.b. eine ressourcenhungrige Wordpress Webseite zu erstellen, da dies aufgrund der existierenden Werkzeuge und Artikel im Internet 
+recht schnell und einfach gehen kann. 
+Wenn man aber an einem Program arbeitet welches mal von allen Menschen verwendet werden soll, macht das schon viel mehr Sinn. 
+Aber ein für mich noch viel wichtiger Punkt an der Stelle wer hostet nachher eigentlich die Communities. 
+Sind es große Firmen die sich fette Server leisten können? Oder sind es kleine Vereine die um jeden Cent kämpfen?
+Sind es Einzelkämpfer die klein-computer wie Raspberry-PI betreiben? 
```

Comment:
I would like to throw in a different angle on this topic:
If we break this comparison down to money, one developer hour is almost the value of 1 month of operating a mid tier server. If the developer invests 20 hours in efficiency optimization it equals a year of server operations.
What I want to say is that the costs of developers vs server hardware are not equal. I agree that resources efficiency is a very important goal, but it can be postponed till we scale worldwide, since then and only then costs for server resources will have an adequat value (1 developer hour will effect many servers and therefore they will become comparable).